### PR TITLE
chore(examples): switch default from yarn to npm in Dockerfiles

### LINKED ISSUE #61293: Docs: Use `npm i -g` Instead of `yarn global add` on Dockerfile Example
### What is the improvement or update you wish to see?

Using `npm i -g pnpm` would be better than `yarn global add pnpm` because `npm` is always installed on the host, but `yarn` might not be depending on the project.

This change could also be applied to the multi-environment examples: https://github.com/vercel/next.js/blob/canary/examples/with-docker-multi-env/docker/development/Dockerfile#L15

### Is there any context that might help us understand?

I tried to use this Dockerfile as a starting point, but it failed when I attempted to build a `pnpm` project (monorepo) with Next as an app.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/pages/building-your-application/deploying#docker-image

### PR DESCRIPTION
**Pull request by samcx**
## Changes

- Switch default package manager to npm
- Fixes https://github.com/vercel/next.js/issues/61293

Closes NEXT-2291

### CODE REVIEW

File: examples/with-docker-compose/next-app/dev.Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
 RUN \
   if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
   elif [ -f package-lock.json ]; then npm ci; \
-  elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
+  elif [ -f pnpm-lock.yaml ]; then npm install -g pnpm && pnpm i; \
```
Comment (styfle) [inline examples/with-docker-compose/next-app/dev.Dockerfile:10]:
```suggestion
  elif [ -f pnpm-lock.yaml ]; then corepack enable pnpm && pnpm i; \
```

**Review by styfle — COMMENTED** [review]
Let's use corepack

https://styfle.dev/blog/corepack

File: examples/with-docker-compose/next-app/dev.Dockerfile (hunk 1 above)
Comment (samcx) [reply examples/with-docker-compose/next-app/dev.Dockerfile]: @styfle Is it `corepack enable pnpm` or just `corepack enable`?

Comment (styfle) [reply examples/with-docker-compose/next-app/dev.Dockerfile]: Depends if you want to enable all package managers or only pnpm

**Review by styfle — APPROVED** [review]

File: examples/with-docker-compose/next-app/dev.Dockerfile (hunk 1 above)
Comment (samcx) [reply examples/with-docker-compose/next-app/dev.Dockerfile]: I guess it makes sense as `corepack enable pnpm` in this case then!

**Pull request merged**